core/src/main/java/org/apache/calcite/avatica/Meta.java

```diff
@@ -647,12 +648,13 @@ public static CursorFactory create(@JsonProperty("style") Style style,
     public static final CursorFactory LIST =
         new CursorFactory(Style.LIST, null, null, null);
 
-    public static CursorFactory record(Class resultClazz) {
-      return new CursorFactory(Style.RECORD, resultClazz, null, null);
```

Comment:
Maybe we should keep the method for a short while for backwards compatibility. WDYT?

Reply:
Agreed, not sure if we should target `2.0` like for the other "deprecated" annotations I have seen around, rather than `1.19` as I have seen in your branch. Both work for me.

Reply:
I am not sure if there we are ever going to have a `2.0` :) As you prefer, I dont mind.

Reply:
Let's leave 1.19 then, by the next release we will have time to hear if there is any concerns from downstream projects.